src/static-template.js

```diff
@@ -14,52 +14,132 @@
  * limitations under the License.
  */
 
+import {TempCache} from './utils/temp-cache';
 import {devAssert} from './log';
+import {
+  getAmpdoc,
+  getServiceForDoc,
+  registerServiceBuilderForDoc,
+} from './service';
 import {map} from './utils/object.js';
 
-let container;
+
+const FLUSH_CACHE_AFTER_MACRO_TASKS = 3;
+const SERVICE = 'html';
+
+
+/** @typedef {function(!Array<string>):!Element} */
+export let HtmlLiteralTagDef;
+
+
+/** @param {!./service/ampdoc-impl.AmpDoc} ampdoc */
+export function installHtmlForDoc(ampdoc) {
+  registerServiceBuilderForDoc(ampdoc, SERVICE, buildHtmlService);
+}
+
+
+/**
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
+ * @return {{
+ *  htmlInternal: !HtmlLiteralTagDef,
+ *  cachedHtmlInternal: !HtmlLiteralTagDef,
+ * }}
+ */
+function getHtml(nodeOrDoc) {
+  return getServiceForDoc(getAmpdoc(nodeOrDoc), SERVICE);
```

Comment:
Wait, didn't we prevent passing an `ampdoc` into a `getServiceForDoc` call? @choumx

Reply:
Looking forward to knowing what I did wrong and fix it 🤓

Reply:
It's incomplete. Most of the getters in services.js are migrated, but not all so we can't change `getServiceForDoc` yet.

But yea, change `nodeOrDoc` to `element`.